### ISSUE TITLE
Setting align property of annotation object causes background to become transparent

### ISSUE THREAD

**Issue by Jstuff36** [highcharts/highcharts#7498]
Whenever I change the `align` property of my annotation object to `left` or `right` the background of my annotation becomes transparent. This does not when it is left at `center`, but for my app I need to use `left` and `right`

#### Expected behaviour
Background does not go away.

#### Actual behaviour
Background goes away.

#### Live demo with steps to reproduce
Here is my annotation object

`      const annotationObject = {
        id: "1",
       labelOptions: {
          useHTML: true,
        },
        labels: [{
          backgroundColor: "rgba(51, 51, 51, 0.75)",
          padding: 10,
          borderColor: "transparent",
          point: {
            x: values[0],
            xAxis: 0
          },
          style: {
            color: "red"
          },
          verticalAlign: "top",
          align: "center",
          text: 'There is my background!'
        }]
      };`

#### Affected browser(s)
Chrome

This is the first time I've raised an issue. How can I probably format my code? Also, how can I attach pictures?

**Comment by pawelfus** [comment]
Hi @Jstuff36 

Thank you for reporting this issue! It's interesting, applying `x+y` values for a point generates the background, with only `x` value - sometimes background is missing, take a look: http://jsfiddle.net/kbjxrfxf/2/

PS: You can attach files in many ways, when submitting an issue or commenting an issue, you can see: 

![screen shot 2017-12-08 at 12 54 31](https://user-images.githubusercontent.com/1453926/33764980-04c7fe8a-dc17-11e7-9894-2c66377db480.png)

Regarding formatting, you can see "Styling with Markdown is supported" - click on it for the details.

**Comment by Jstuff36** [comment]
Hey @pawelfus 

Yea that's another way to reproduce the issue. I did some investigation and was able to pinpoint the problem to more or less how highcharts draws SVG. I don't 100% understand why or how, but it seems that when giving the annotation a position such as `align: left` along with setting the `useHTML` property to `true`, highcharts will sometimes miscalculate the width of the background and draw but otherwise times it will not. I could not figure out why it sometimes calculates a background width and other times it does not. I was able to around this issue by setting and all the properties of the background to `transparent` and then doing all my styling with the inline css that highcharts supports. I hope this helps someone.
